DataFormats/L1THGCal/interface/HGCalMulticluster.h

```diff
@@ -49,12 +51,27 @@ namespace l1t {
       return math::PtEtaPhiMLorentzVector(pt() * interpretationFraction(eInt), eta(), phi(), 0.);
     }
 
-    std::vector<EnergyInterpretation> energyInterpretations() const {
-      std::vector<EnergyInterpretation> keys;
-      for (auto &kvp : energyInterpretationFractions_) {
-        keys.emplace_back(kvp.first);
+  private:
+    template <typename Iter>
+    struct KeyGetter : std::unary_function<typename Iter::value_type, typename Iter::value_type::first_type> {
+      const typename Iter::value_type::first_type& operator()(const typename Iter::value_type& p) const {
+        return p.first;
       }
-      return keys;
+    };
+
+    template <typename Iter>
+    boost::transform_iterator<KeyGetter<Iter>, Iter> key_iterator(Iter itr) const {
+      return boost::make_transform_iterator<KeyGetter<Iter>, Iter>(itr, KeyGetter<Iter>());
+    }
+
+  public:
+    typedef boost::transform_iterator<KeyGetter<std::map<EnergyInterpretation, double>::const_iterator>,
+                                      std::map<EnergyInterpretation, double>::const_iterator>
+        EnergyInterpretation_const_interator;
```

Comment:
Typo: `EnergyInterpretation_const_iterator` instead of `EnergyInterpretation_const_interator`